src/openpodcast.js

```diff
@@ -1,25 +1,39 @@
+const runtime = window.browser || chrome
+
 //insert html at the beginning of the website
 function showCredentialsOnWebsite(creds) {
-  let message = "No Spotify credentials found, are you logged in on podcasters.spotify.com? (reload needed)"
+  let message = "No Spotify credentials found, are you logged in on podcasters.spotify.com? (maybe reload this page)"
   if (creds.sp_key && creds.sp_dc) {
-    message = `Spotify Credentials: key: ${creds['sp_key']}, dc: ${creds['sp_dc']}`
+    message = `Spotify keys: key: ${creds['sp_key']}, dc: ${creds['sp_dc']} END`
   }
-  const credsHTMLElement = document.createTextNode(message);
-  document.body.prepend(credsHTMLElement)
+  document.getElementById("openpodcast-plugin").innerHTML = message
 }
 
 //communicate with background script to fetch cookies (only background script is allowed to do this)
 async function retrieveSpotifyCookies() {
+  console.log("retrieving spotify keys from service worker")
   // sends a message to the background.js and retrieve cookies
-  const message = await browser.runtime.sendMessage({type: "spotifycookies"})
+
+  const message = await chrome.runtime.sendMessage({ type: "spotifycookies" })
+
   const cookies = message.response
+  console.log(cookies)
   const spotifyCreds = cookies
     //filter what cookies we need and map to a simple lookupobject
     .filter((cookie) => cookie.name === "sp_key" || cookie.name === "sp_dc")
-    .reduce((result, cookie) => { result[cookie.name]=cookie.value; return result; },{})
+    .reduce((result, cookie) => { result[cookie.name] = cookie.value; return result; }, {})
   //show credentials in the HTML code of the OpenPodcast website
   showCredentialsOnWebsite(spotifyCreds)
 }
 
-//init
-retrieveSpotifyCookies()
\ No newline at end of file
+// check if current website is the OpenPodcast website where we want to show the credentials
+function isOnOpenPodcastWebsite() {
+  return window.location.href.match(/https:\/\/openpodcast.dev\/.*connect/) && document.getElementById("openpodcast-plugin")
+}
+
+//wait 1sec until the website is loaded (maybe a problem with docsify) and then check if we are on the right website
```

Comment:
docsify can take multiple seconds to load from my experience. Maybe we want to increase the waiting time to 2-3 seconds?

Reply:
it is a loop anyways